app/scripts/controllers/swaps.js

```diff
@@ -101,19 +110,35 @@ export default class SwapsController {
     this._setupSwapsLivenessFetching()
   }
 
+  // Sets the refresh rate for quote updates from the MetaSwap API
+  async _setSwapsQuoteRefreshTime() {
+    let refreshTime
+    try {
+      refreshTime = await this._fetchSwapsQuoteRefreshTime()
+    } catch (e) {
+      refreshTime = FALLBACK_QUOTE_REFRESH_TIME
+      console.error('Request for swaps quote refresh time failed: ', e)
+    }
+    this.setSwapsQuoteRefreshTime(refreshTime)
+  }
+
   // Once quotes are fetched, we poll for new ones to keep the quotes up to date. Market and aggregator contract conditions can change fast enough
   // that quotes will no longer be available after 1 or 2 minutes. When fetchAndSetQuotes is first called it, receives fetch that parameters are stored in
   // state. These stored parameters are used on subsequent calls made during polling.
   // Note: we stop polling after 3 requests, until new quotes are explicitly asked for. The logic that enforces that maximum is in the body of fetchAndSetQuotes
   pollForNewQuotes() {
+    const {
+      swapsState: { swapsQuoteRefreshTime },
+    } = this.store.getState()
+
     this.pollingTimeout = setTimeout(() => {
       const { swapsState } = this.store.getState()
       this.fetchAndSetQuotes(
         swapsState.fetchParams,
         swapsState.fetchParams?.metaData,
         true,
       )
-    }, QUOTE_POLLING_INTERVAL)
+    }, swapsQuoteRefreshTime - QUOTE_POLLING_DIFFERENCE_INTERVAL)
```

Comment:
Won't this "fetch and set quotes" 10 seconds sooner than the refresh time would dictate? I don't follow what this difference interval is for.

Reply:
This maintains what we have now -- I'm happy to create another patch when Dan comes back and we know more.

Reply:
Oh, I hadn't realized! I'll give this another look.